src/child.js

```diff
@@ -94,14 +100,14 @@ class Child {
 
     // Expecting JSON data
     try {
-      actualData = JSON.parse(dataReceived);
+      actualData = this.parse(dataReceived);
```

Comment:
I went this direction because the `config` is merged into the `this` object on instantiation (`Object.assign(this, config)`), however I noticed a mixture of either `this.{configKey}` and `this.config.{configKey}`, so if you'd rather I'd change to the latter I can.

Reply:
I got your point of doing this but let's change it to latter i.e. `this.config.parse` and `this.config.stringify`.
This way it looks obvious that we are using the user-provided methods.